vector/src/main/java/im/vector/app/features/home/room/detail/TimelineFragment.kt

```diff
@@ -1578,12 +1578,21 @@ class TimelineFragment @Inject constructor(
                 views.hideComposerViews()
                 views.notificationAreaView.render(NotificationAreaView.State.Tombstone(mainState.tombstoneEvent))
             }
-        } else if (summary?.membership == Membership.INVITE && inviter != null) {
+        } else if (summary?.membership == Membership.INVITE && inviter != null && !timelineArgs.isInviteAlreadyAccepted ) {
             views.hideComposerViews()
             lazyLoadedViews.inviteView(true)?.apply {
                 callback = this@TimelineFragment
                 isVisible = true
-                render(inviter, VectorInviteView.Mode.LARGE, mainState.changeMembershipState)
+                render(inviter, VectorInviteView.Mode.LARGE, mainState.changeMembershipState, false)
+                setOnClickListener(null)
+            }
+            Unit
+        } else if (summary?.membership == Membership.INVITE && inviter != null && timelineArgs.isInviteAlreadyAccepted) {
+            views.hideComposerViews()
+            lazyLoadedViews.inviteView(true)?.apply {
+                callback = this@TimelineFragment
+                isVisible = true
+                render(inviter, VectorInviteView.Mode.LARGE, mainState.changeMembershipState, true)
                 setOnClickListener(null)
             }
```

Comment:
I don't see many differences between these two blocs, it seems that there is only the third parameter in the `render` function which depends on the `isInviteAlreadyAccepted`. I think you can keep the previous condition and add the new condition inside the render method params (create a variable if you think it is more clear).

Also, do not hesitate to convert the `if` to a `when` statement if you add more cases.

Reply:
Already fixed

Reply:
yes, you're right :)